### PR TITLE
Version 5.2

### LINKED ISSUE #1697: K version bump feature list (5.2.0)
- Features:
  - Deprecated `imports syntax`: #1851 
  - Constant folding kompiler pass: #1896, #1946, #1958
  - MAP module is automatically imported into all modules that declare configurations: #1965 
  - `krun --search` supported in LLVM backend: #1978 
  - #2082
  - kprovex: #2089
  - #1658
  - #2164
  - #2199
  - #2204
  - #2209
  - #2214
  - #2218
  - #2002
  - #1584
  - #2292
- Documentation updates:
  - #1706 
  - #1777 
  - K tutorial through lesson 12: #1847 #1973 #1992 #1993 #1994 #2005 #2011 #2018 #2023
  - #1926 
  - #1930
  - #1995
  - #1684 
  - #1968 
  - #1955
  - #2054
  - #2059
  - #2127
- Bug fixes:
  - #1906 
  - #1088 
  - #1940 
  - #1990 
  - #1967 
  - #1969 
  - #2001
  - #2071
  - #2092
  - #2102
  - #2123
  - #2153
  - #2160 
  - #2161
  - #2172
  - #2184
  - #2203
  - #2226
  - #2266
  - #2280
- Performance improvements:
  - #1944 
  - #1952
  - #1996
  - #2060
  - #2012
  - #2271
- Dependencies
  - Switched to java serialization: #2050 #2051
  - LLVM version 10: #2049, #2213
  - Scala: #2189

Check:

- [ ] File `./package/version` has been bumped to `NEW_MAJOR.NEW_MINOR.0`, and `./package/version.sh sub` has been invoked.
- [ ] `CHANGELOG.md` has been updated with new features.
- [ ] `krun` help script is correctly reflecting the available options
- [ ] All `*Options.java` files are up-to-date.
- [ ] K active projects page is up-to-date (receiving commits in the last 3 months).
- [ ] Audit of warnings in regression test-suite and tutorial.
- [ ] Audit package dependencies in `package/`.

### PR DESCRIPTION
**Pull request by ehildenb**
To Check:

- [x] File `./package/version` has been bumped to `NEW_MAJOR.NEW_MINOR.0`, and `./package/version.sh sub` has been invoked.
- [x] `CHANGELOG.md` has been updated with new features.
- [x] K active projects page is up-to-date (receiving commits in the last 3 months).
- [x] Audit of warnings in regression test-suite and tutorial (inspect CI output for unintentional warnings).
- [x] Audit package dependencies in `package/` (look at various package/dependency lists in `package/` subdirectory and think whether any of them is now not needed).

### CODE REVIEW

**Comment by radumereuta** [comment]
Link to the haskell backend changelog.

**Comment by radumereuta** [comment]
@ehildenb please review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,112 @@
...
+- The option `--help-experimental` or `-X` is removed from all tools, and 
+  all the `--help` option lists are flattened.
+
+- Rename `pending-documentation.md` into `USER_MANUAL.md`
+
+Misc/Bug Fixes
+--------------
+
+- Modules marked as `not-lr1` are automatically excluded from the generated
+    Bison parser, instead of causing an error.
```
Comment (ehildenb) [inline CHANGELOG.md]: Inconsistent formatting here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,112 @@
...
+
+- Formatting of several error messages is improved, especially indentation.
+
+- File symlink handling in `requires` statements is improved.
+
+- `kast` tool no longer silently ignores the `--output-file` parameter.
+
+- Verbose mode offers more information about timings and scanners being created.
+
+- Added `-I`, `--md-selector` and `--no-prelude` to kprove, kprovex and kbmc
```
Comment (ehildenb) [inline CHANGELOG.md]:
```suggestion
- Added `-I`, `--md-selector` and `--no-prelude` to `kprove`, `kprovex` and `kbmc`.
```

**Comment by ehildenb** [comment]
Looks good to me. Make sure to look through the checklist at the top of the PR @radumereuta . I've trimmed it down.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,112 @@
...
+- `kompile` will now attempt to simplify constant expressions which appear
+  in the RHS of rules.
+
+- The LLVM backend now supports the non-deterministic `search` feature.
+  This must be enabled at `kompile` time with `--enable-search --backend llvm`
+  and only supports the `--search|--search-final` options.
+
+- The sentence type `imports syntax` has been removed. It was only used in one
+  place in the tutorial, which was easy to resolve. This allows simplifying
+  several code-paths in `kompile`.
```
Comment (Baltoli) [inline CHANGELOG.md:18]: -> allows several code paths in `kompile` to be simplified

Comment (Baltoli) [inline CHANGELOG.md]: -> allows several code paths in `kompile` to be simplified

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,112 @@
...
+- Added `kprovex`. A lightweight version of `kprove` which restricts the spec files
+  to claims and token syntax. This avoids several bugs in the old code path and
+  slightly improves the front end overhead. To convert the old tests, you need to
+  kompile ahead of time the definition with the simplification rules and extra
+  helper syntax needed for the proofs.
+
+Documentation
+-------------
+
+- Relicence as BSD 3-Clause
```
Comment (Baltoli) [inline CHANGELOG.md]: -> relicense

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,112 @@
...
+
+-   Haskell backend is updated to version d4239d97f5b53188460e7399b683111b5687661c.
+
+-   LLVM backend is updated to version 23c2adcd21a1486442062ecc4ae051548215bcc3.
+
+-   K Web Theme is updated to version b458d1461c31760024ea06e0e50f25806ace5e2c.
+
+-   JavaCC is updated to version 4.1.4.
+
+-   commons-io Java library is bumped to version 2.7.
```
Comment (Baltoli) [inline CHANGELOG.md]: The indentation here on list entries is different to above - is this deliberate?

**Review by Baltoli — CHANGES_REQUESTED** [review]

**Review by Baltoli — APPROVED** [review]
Looks good!

**Comment by ehildenb** [comment]
LGTM! I can't approve because I opened this one a long time ago.

**Pull request merged**